### PR TITLE
Improved tests: illegal arguments

### PR DESCRIPTION
**Pull request by evanfields**
Improve tests:
* more extensive comparison of Julia and Python package outputs
* test that various illegal arguments throw an exception

Corresponding code changes:
* Better handling of illegal arguments
* Avoid integer overflow

### CODE REVIEW

File: src/GeohashHilbert.jl [hunk 1]
```diff
@@ -108,15 +108,30 @@ string.
 - `lat::Real`: latitude of point to encode. Must be in [-90,90].
 - `precision::Integer`: precision of returned geohash. Must be positive.
 - `bits_per_char ∈ [2,4,6]`: how many bits of information each character encodes.
+
+!!! note
+    To avoid overflow in machine arithmetic, `precision * bits_per_char` must be <= 62.    This still allows meter-level granularity. If your use case requires sub-meter
```
Comment (jdreben) [inline src/GeohashHilbert.jl]: Should this be split into two lines?

Comment (evanfields) [reply src/GeohashHilbert.jl]: yes whoops. Atom wraps lines so prettily I didn't notice.

File: test/runtests.jl [hunk 2]
```diff
@@ -64,33 +64,46 @@ end
...
-        [180, 47, 5, "22333"],
-        [180, 90, 7, "2222222"],
-        [13, 47, 21, "210032131130033230313"],
-        [-13, 47, 21, "123301202203300103020"],
-        [13, -47, 21, "321103020023322121020"],
-        [-13, -47, 21, "012230313310011212313"]
-    ]
-    for tup in llp_g
-        @test GHH.encode(tup[1:3]...) == tup[4]
+    # python_hashes.csv has columns lon, lat, prec, bits_per_char, geohash
```
Comment (jdreben) [inline test/runtests.jl]: nit: bpc not bits_per_char, though almost clearer to spell it out here

**Review by jdreben — APPROVED** [review]
👍

**Pull request merged**